pandas/core/window/online.py

```diff
@@ -0,0 +1,111 @@
+from typing import (
+    Dict,
+    Optional,
+)
+
+import numpy as np
+
+from pandas.compat._optional import import_optional_dependency
+
+from pandas.core.util.numba_ import (
+    NUMBA_FUNC_CACHE,
+    get_jit_arguments,
+)
+
+
+def generate_online_numba_ewma_func(engine_kwargs: Optional[Dict[str, bool]]):
+    """
+    Generate a numba jitted groupby ewma function specified by values
+    from engine_kwargs.
+    Parameters
+    ----------
+    engine_kwargs : dict
+        dictionary of arguments to be passed into numba.jit
+    Returns
+    -------
+    Numba function
+    """
+    nopython, nogil, parallel = get_jit_arguments(engine_kwargs)
+
+    cache_key = (lambda x: x, "online_ewma")
+    if cache_key in NUMBA_FUNC_CACHE:
+        return NUMBA_FUNC_CACHE[cache_key]
+
+    numba = import_optional_dependency("numba")
+
+    @numba.jit(nopython=nopython, nogil=nogil, parallel=parallel)
+    def online_ewma(
+        values: np.ndarray,
+        deltas: np.ndarray,
+        minimum_periods: int,
+        old_wt_factor: float,
+        new_wt: float,
+        old_wt: np.ndarray,
+        adjust: bool,
+        ignore_na: bool,
+    ):
+        result = np.empty(values.shape)
+        weighted_avg = values[0]
+        nobs = (~np.isnan(weighted_avg)).astype(np.int64)
+        result[0] = np.where(nobs >= minimum_periods, weighted_avg, np.nan)
+
+        for i in range(1, len(values)):
+            cur = values[i]
+            is_observations = ~np.isnan(cur)
+            nobs += is_observations.astype(np.int64)
+            for j in range(len(cur)):
+                if not np.isnan(weighted_avg[j]):
```

Comment:
can you add blank lines before comments. add some comments on L51 on what is happening here